Add functionality to deal with spectral similarity

LiesaSalzer opened the pull request:
Just as discussed with @tnaake I implemented following steps: 

1.  calculation and adding spectral similarity matrices to the structural `AdjacencyMatrix` 
2.  this object can be thresholded by `threshold(am, type = "threshold")` and different filtering options that is defined by the user 
3.  the thresholded `AdjacencyMatrix` can be combined by combine() with the thresholded statistical `AdjacencyMatrix`
4. Updated documentation of functions
5. Added toy example file for MS2 similarity
6. Examples and description in vignette 
7. Unit tests 

Hope everything works for you now!

LiesaSalzer committed:
add ms2 test data

LiesaSalzer committed:
add spectral similarity

LiesaSalzer committed:
minor change in AllClasses

LiesaSalzer committed:
adapt threshold

LiesaSalzer committed:
change threshold of structural adjacencyMatrix --> stores results in "binary" instead of "consensus"

LiesaSalzer committed:
remove import MsCoreUtils

LiesaSalzer committed:
add ndotproduct import from MsCoreUtils

LiesaSalzer committed:
fix importFrom

LiesaSalzer committed:
add @importFrom Spectra spectraVariables

LiesaSalzer committed:
update documentation and unit tests

LiesaSalzer committed:
minor issue in unit test

tnaake commented on NAMESPACE:
```diff
@@ -1,6 +1,7 @@
 # Generated by roxygen2: do not edit by hand
 
 export(AdjacencyMatrix)
+export(addSpectSimil)
```
could we have a proper name here, rather than the abbreviated one, e.g. `addSpectralSimilarity`?

tnaake commented on R/structural.R:
```diff
@@ -474,3 +474,142 @@ rtCorrection <- function(am, x, transformation, var = "group") {
+  }
+  
+  
+ return(am_structural)
```
would it be possible to have only the `adj_spec` object passed to the `addSpectralSimilarity` function? I think it's a bit superfluous to have the `compareSpectra` step in this function as this can also be done outside of `MetNet`. In that case the function would be shorter, more easier to understand, less work to maintain, and we reduce the number of dependencies. In the end it also doesn't matter if the MS2 similarity matrix is created via `Spectra` or some other package. 

We could call the argument `ms2_similarity` (this could be a named list with entries specifying the name of the assay, e.g. `list(ndotproduct = matrix(...), dotproduct = matrix(...)`.

tnaake replied:
`Spectra` could go into `Suggests` in the DESCRIPTION.

tnaake commented on vignettes/MetNet.Rmd:
```diff
@@ -376,6 +376,73 @@ rowData(struct_adj) <- x_annotation
+spectra consolidation has to be performed first (e.g. using the function 
+`Spectra::combineSpectra()`).
+
+Features from `spectra` will by matched accordingly to the feature names in the 
```
Typo: will be

tnaake commented on tests/testthat/test_spectSimil.R:
```diff
@@ -0,0 +1,98 @@
+
+
+test_that("threshold and combine similarity", {
+  expect_error(threshold(spect_adj, type = "top1", args = list(n = 2)), 
```
please use throughout Bioc coding style. FOr instance, spaces properly (see e.g. `c("binary","group","formula","mass","ndotproduct")` and intendations of 4.

tnaake commented on vignettes/MetNet.Rmd:
````diff
@@ -376,6 +376,73 @@ rowData(struct_adj) <- x_annotation
+```
+
+
+Further, the spectral similarity matrix can be thresholded using the function threshold. 
````
should be 'function `threshold`'

tnaake commented on vignettes/MetNet.Rmd:
```diff
@@ -376,6 +376,73 @@ rowData(struct_adj) <- x_annotation
+
+Further, the spectral similarity matrix can be thresholded using the function threshold. 
+The user needs to define wheather the data should be thresholded or not 
+If multiple methods have been used to generate the spectral matrices, different threshold parameters can be defined (for detailed description see thresholding of statstical)
```
statistical

tnaake commented on tests/testthat/test_spectSimil.R:
```diff
@@ -0,0 +1,98 @@
+                                        "formula","mass","ndotproduct"))
+  expect_equal(sum(assay(spect_adj, "ndotproduct"), na.rm = TRUE), 
+               45.16721, tolerance = 5e-01)
+  expect_equal(as.vector(assay(spect_adj, "ndotproduct")[7,1:10]), 
```
for testing purposes a MS2 similarity matrix can be stored as an `RData` object in `MetNet` to avoid dependency on `Spectra`. There can be then a small script on how this matrix was created.

tnaake commented on R/statistical.R:
```diff
@@ -953,20 +953,24 @@ getLinks <- function(mat, exclude = "== 1", decreasing = TRUE) {
 #'
 #' @aliases threshold
```
the function `threshold` should be moved to `combine.R` as it is used now by both `statistical` and `structural`.

tnaake reviewed (changes requested):
Hi @LiesaSalzer 

very good! Some small changes only that should not require too much adjustments (mainly removing the calculation of MS2 similaritites within the `addSpectralSimilarity` function).

Many thanks for the great work! 

Best,
Thomas

LiesaSalzer committed:
updates in spectral similarity

LiesaSalzer committed:
update on spectral similarity

Merge branch 'devel_liesa' of https://github.com/LiesaSalzer/MetNet into devel_liesa

# Conflicts:
#	R/structural.R

LiesaSalzer committed:
- change addSpectSimil to addSpectralSimilarity
- change addSpectralSimilarity to use similarity matrices as input
- add test data and description
- update description
- update vignette
- update unit tests

LiesaSalzer commented:
Hi @tnaake, 
Thanks for your feedback! 
I try to address all your comments in the last commit. 
I think it should be fine now :)

tnaake commented:
Many thanks, for the PR :) Looks great @LiesaSalzer !
